Add function `fromDaysSinceYearZero()`

jh0x opened the pull request:
Closes #55308 

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added function `fromDaysSinceYearZero()` which is similar to MySQL's `FROM_DAYS`. E.g. `SELECT fromDaysSinceYearZero(739136)` returns `2023-09-08`.

jh0x committed:
Stub of function fromDaysSinceYearZero

jh0x committed:
Flesh out logic for fromDaysSinceYearZero and tests

robot-clickhouse commented:
<!-- automatic status comment for PR #56088 from jh0x/ClickHouse:feature-from-days-zero -->
*This is an automated comment for commit f4dc2548a41196826b604560f52c538beb54f808 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56088/f4dc2548a41196826b604560f52c538beb54f808/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>❌ failure</td></tr>
<tbody>
</table>

jh0x committed:
Fix failing checks

jh0x committed:
Fix typos, add to/fromDaysSinceYearZero to fuzz

jh0x committed:
More fixes after CI, negatives check

rschu1ze committed:
Use FunctionArgumentDescriptors in getReturnTypeImpl()

rschu1ze committed:
Remove a few unnecessary function implementations

rschu1ze committed:
Uppercase constants

rschu1ze committed:
Update docs

rschu1ze committed:
Move DAYS_BETWEEN_YEARS_0_AND_1970 back to indicate the relation to ToDaysSinceYearZero

rschu1ze committed:
Remove overflow setting

Reason 1:
- https://github.com/ClickHouse/ClickHouse/pull/55696#issuecomment-1774130793
  Basically: Setting `date_time_overflow_behavior` applies only to
  conversions between Date <-> Date32 <-> DateTime <-> DateTime64 but
  it does not control the behavior of functions.

Reason 2:
- It's overkill (even if well-intented overkill). ClickHouse generally
  has a garbage in - garbate out philosophy, so that should be the
  default behavior which corresponds to behavior "ignore". Moreover,
  function 'fromDaysSinceYearZero()' mostly exists for compatibility /
  parity with MySQL and MySQL takes the same stance.

rschu1ze committed:
Add FROM_DAYS() alias

rschu1ze committed:
Make the tests 14.3% more aesthetic

rschu1ze committed:
Remove more dead code

rschu1ze commented:
@jh0x Thanks a lot for this contribution. I took the freedom to make some adjustments (please see my individual commits). The main change is this: https://github.com/ClickHouse/ClickHouse/pull/56088/commits/0b587d3ceab429bb5e85c6dfaf3d01943097495a But this is totally not to diminish your effort, you'll be mentioned in the changelog!

rschu1ze committed:
Fix style

rschu1ze committed:
Fix FastTest
